AST Check Disagnostics broken by zig master

@Techatrix  The assumption noted in `getAstCheckDiagnostics` is no longer true in zig master. https://github.com/zigtools/zls/blob/6358c5b23e38a6d4a50be7e043bb9fdc782f39d6/src/Server.zig#L367

Due to https://github.com/ziglang/zig/commit/a99d465f624d9cdd82d270e03dd931c0539df8d7 all `--color off` does now is disable color other than that the format of the error message is the same which breaks the parsing code.

## Example

```zig
const std = @import("std");

pub fn main() void {
    doesnt.exist();
}
```
Results in:
```
$ zig ast-check --color off src/main.zig
src/main.zig:4:5: error: use of undeclared identifier 'doesnt'
    doesnt.exist();
    ^~~~~~
```
Which causes zls to:
```
thread 49931 panic: attempt to use null value
/home/lee/src/zls/src/Server.zig:378:76: 0x2f0569 in Server.getAstCheckDiagnostics (zls)
            .line = (try std.fmt.parseInt(u32, pos_and_diag_iterator.next().?, 10)) - 1,
                                                                           ^
/home/lee/src/zls/src/Server.zig:199:35: 0x2ed1bb in Server.publishDiagnostics (zls)
        try getAstCheckDiagnostics(server, handle, &diagnostics);
                                  ^
/home/lee/src/zls/src/Server.zig:1755:34: 0x2ce756 in Server.openDocumentHandler (zls)
    try server.publishDiagnostics(writer, handle);
                                 ^
/home/lee/src/zls/src/Server.zig:2490:57: 0x2c69f8 in Server.processJsonRpc (zls)
                        done = extractErr(method_info[2](server, writer, id, request_obj));
                                                        ^
/home/lee/src/zls/src/main.zig:50:34: 0x2aea5d in loop (zls)
        try server.processJsonRpc(writer, buffer);
                                 ^
/home/lee/src/zls/src/main.zig:280:13: 0x2a94f8 in main (zls)
    try loop(&server);
            ^
/home/lee/zig/0.10.0-dev.4474+b41b35f57/files/lib/std/start.zig:578:37: 0x27c420 in std.start.posixCallMainAndExit (zls)
            const result = root.main() catch |err| {
                                    ^
/home/lee/zig/0.10.0-dev.4474+b41b35f57/files/lib/std/start.zig:340:5: 0x27c1d2 in std.start._start (zls)
    @call(.{ .modifier = .never_inline }, posixCallMainAndExit, .{});
    ^
```

--- a/src/Server.zig
+++ b/src/Server.zig
@@ -368,6 +368,8 @@ fn getAstCheckDiagnostics(
     var line_iterator = std.mem.split(u8, stderr_bytes, "\n");
 
     while (line_iterator.next()) |line| lin: {
+        if (!std.mem.startsWith(u8, line, "<stdin>")) continue;
+
         var pos_and_diag_iterator = std.mem.split(u8, line, ":");
         const maybe_first = pos_and_diag_iterator.next();
         if (maybe_first) |first| {
@@ -1534,7 +1536,7 @@ fn initializeHandler(server: *Server, writer: anytype, id: types.RequestId, req:
     // NOTE: everything is initialized, we got the client capabilities
     // so we can now format the prebuilt builtins items for labelDetails
     if (server.client_capabilities.label_details_support) {
-        for(server.builtin_completions.items) |*item| {
+        for (server.builtin_completions.items) |*item| {
             try formatDetailledLabel(item, std.heap.page_allocator);
         }
     }

